src/main/java/com/amazon/opendistroforelasticsearch/sql/query/AggregationQueryAction.java

```diff
@@ -236,28 +236,38 @@ public SqlElasticSearchRequestBuilder explain() throws SqlParseException {
 
     private AggregationBuilder getGroupAgg(Field groupByField, Select select) throws SqlParseException {
         AggregationBuilder lastAgg = null;
+        Field shadowField = null;
+
         for (Field selectField : select.getFields()) {
             if (selectField instanceof MethodField && selectField.getName().equals("script")) {
                 MethodField scriptField = (MethodField) selectField;
                 for (KVValue kv : scriptField.getParams()) {
                     if (kv.value.equals(groupByField.getName())) {
-                        lastAgg = aggMaker.makeGroupAgg(scriptField);
+                        shadowField = scriptField;
                         break;
                     }
                 }
             }
         }
 
-        if (lastAgg == null) {
+        if (shadowField == null) {
             for (Field selectField: select.getFields()) {
```

Comment:
Is it possilbe to combine this forloop with the forloop on L241. For me, the if null check of shadowFiled is a little bit complex.

Reply:
I'm not sure whether it was two loops at the beginning, I prefer keep it as is to avoid breaking any logic.